### PR TITLE
[ML] add deployed native models to inference_stats in trained model stats response

### PR DESCRIPTION
**Pull request by benwtrent**
This adds a valid `inference_stats` section for deployed native models.

`inference_stats` is effectively a sub-set of the `deployment_stats`. It's a high level view of the overall stats of the model, deployment_stats contains more detailed information around types of errors seen, throughput, etc.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/AssignmentStats.java [hunk 1]
```diff
@@ -437,6 +438,23 @@ public AssignmentStats setReason(String reason) {
         return this;
     }
 
+    /**
+     * @return The overall inference stats for the model assignment
+     */
+    public InferenceStats getOverallInferenceStats() {
+        return new InferenceStats(
+            0L,
+            nodeStats.stream().filter(n -> n.getInferenceCount().isPresent()).mapToLong(n -> n.getInferenceCount().get()).sum(),
```
[the remark this answers is not in the document]
Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/AssignmentStats.java:447]: It actually isn't a duplicate as it is the sum of inferences across all nodes. I think that's useful

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Comment by dimitris-athanasiou** [comment]
Btw, I think a unit test that checks we sum the node inference counts and failures might be useful to add.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/AssignmentStats.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidkyle) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/AssignmentStats.java:447]: It is a duplicate but I don't think that's a problem. AssignmentStats collects various counts across all nodes

https://github.com/elastic/elasticsearch/blob/68c51f3ada53d9f67411aa9bdb06e49cbb2d2f64/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/AssignmentStats.java#L467

**Review by davidkyle — APPROVED** [review]
LGTM

This fixes the discrepancy where DFA models have inference stats but NLP models do not?

**Comment by benwtrent** [comment]
> This fixes the discrepancy where DFA models have inference stats but NLP models do not?

correct @davidkyle

**Pull request merged**